lib/project_types/extension/models/specification_handlers/default.rb

```diff
@@ -42,38 +42,22 @@ def valid_extension_contexts
           []
         end
 
-        def choose_port?(context)
-          cli_compatibility(context).accepts_port?
+        def choose_port?(_context)
+          false
         end
 
-        def establish_tunnel?(context)
-          cli_compatibility(context).accepts_tunnel_url?
+        def establish_tunnel?(_context)
+          false
         end
 
-        def serve(context:, port:, tunnel_url:)
-          Features::ArgoServe.new(specification_handler: self, cli_compatibility: cli_compatibility(context),
-          context: context, port: port, tunnel_url: tunnel_url).call
+        def serve(context:, **_args)
+          Tasks::Serve.new(context: context).call
```

Comment:
created a default serve command, so ArgoServe would only be used with Argo extensions. Open to naming this something else if `Serve` is too generic / confusing.